custom_components/localtuya/cover.py

```diff
@@ -207,87 +416,85 @@ def available(self):
     @property
     def supported_features(self):
         """Flag supported features."""
-        supported_features = (
-            SUPPORT_OPEN | SUPPORT_CLOSE | SUPPORT_STOP | SUPPORT_SET_POSITION
-        )
+        supported_features = SUPPORT_OPEN | SUPPORT_CLOSE | SUPPORT_STOP | SUPPORT_SET_POSITION
         return supported_features
 
-    @property
-    def current_cover_position(self):
-        # self.update()
-        # state = self._state
-        #        _LOGGER.info("curr_pos() : %i", self._position)
-        # print('curr_pos() : state [{}]'.format(state))
-        return self._position
-
     @property
     def is_opening(self):
```

Comment:
This method can be simplified to:

```python
return self._last_movement == "Opening"
```

Is `"Opening"` something that can vary by the way, so it should be a configuration option?

Reply:
I am not sure if "opening" or "closing" vary. We need information from more cover devices.

**I am going to remove is_opening() and is_closing() for now.** These properties are still unfinished in their implementation. I'm going to work on this in a future PR. There are two types of cover devices we have identified:

(1) Devices like mine that report a wide variety of dps and values, such as the current position, and the last movement of the device ("opening" vs "closing"). For these devices, you can know what the last movement was ("opening" vs "closing") by looking at the dps values. However, you can not definitively know by looking at the dps output whether the device is **currently** "opening" or "closing". We will have to hack together a solution. For example, let's imagine a scenario where the current position of the device is 100 (fully closed) and the last movement is ("closing"). Someone then sets the position to 50. That will immediately change last movement from "closing" to "opening". We could then set is_opening to TRUE until such time as current_cover_position == last_position_set, at which point we set is_opening to FASLE. If someone simply calls open_cover or close_cover, then you can set is_opening or is_closing respectively to TRUE--if someone then calls stop, you know when to stop the is_opening or is_closing. If someone does not calls stop, and lets the cover over or close until it is fully open/fully closed, then we can do some sort of check to see when the current_cover_position updates to 0 or 100. And then there's the edge case, which is what happens when someone controls the cover outside of Home Assistant, using the physical buttons on the device or the device's remote control. I need to investigate how controlling the device in those scenarios modifies dps values (if at all).

(2) Devices like @rospogrigio's that do not let you set or get the position. In essence, all these devices let you do is set them to `on/open` or `off/close` or `stop`. These devices also don't report `opening` or `closing`. For these devices, there's an added complexity, which is that if you open/close and let run until fully open/fully closed, then you can't get the current position to know when it's finished opening/closing. One way around this, perhaps, is to add an optional variable to device setup for number of seconds to fully open/fully close, and we can use that as a max value to determine when the device must have finished opening/closing when a stop was not called.

Reply:
This sounds rather complex to implement in a reliable way, especially to handle all odd-cases and variations. Add to that that we have a poll time of one minute (or is it 30s?). So we have a looong feedback loop before we know what is actually happening. As you say, might be good to do that logic in another PR.

Reply:
I'm unsure if it will be reliable, but i'm going to give it a go. In a different PR.